File: src/main/java/com/kyhsgeekcode/minecraft_env/ObservationSpace.java
package com.kyhsgeekcode.minecraft_env;

public class ObservationSpace {
    private String image;
    private double x;
    private double y;
    private double z;

    private boolean isDead;

    public ObservationSpace(String image, double x, double y, double z, boolean isDead) {
        this.image = image;
        this.x = x;
        this.y = y;
        this.z = z;
        this.isDead = isDead;
    }

    public String getImage() {
        return image;
    }

    public double getX() {
        return x;
    }

    public double getY() {
        return y;
    }

    public double getZ() {
        return z;
    }

    public boolean isDead() {
        return isDead;
    }
}

File: src/main/java/com/kyhsgeekcode/minecraft_env/ObservationSpace.kt
package com.kyhsgeekcode.minecraft_env

import net.minecraft.item.Item

data class ItemStack(
    val rawId: Int,
    val translationKey: String,
    val count: Int,
    val durability: Int,
    val maxDurability: Int
) {
    constructor(itemStack: net.minecraft.item.ItemStack) : this(
        Item.getRawId(itemStack.item),
        itemStack.item.translationKey,
        itemStack.count,
        itemStack.maxDamage - itemStack.damage,
        itemStack.maxDamage
    )
}

data class ObservationSpace(
    val image: String = "",
    val x: Double = 0.0, val y: Double = 0.0, val z: Double = 0.0,
    val yaw: Double = 0.0, val pitch: Double = 0.0,
    val health: Double = 20.0,
    val foodLevel: Double = 20.0,
    val saturationLevel: Double = 0.0,
    val isDead: Boolean = false,
    val inventory: List<ItemStack> = listOf(),
)